ycmd.el

```diff
@@ -254,6 +254,26 @@ use `ycmd-parse-buffer'."
   :group 'ycmd
   :type 'boolean)
 
+(defcustom ycmd-url-proxy-services nil
+  "Specify proxies for local traffic with the ycmd server.
+
+Sets the variable `url-proxy-services' in `ycmd--request' when
+sending requests to the ycmd server:
+
+nil
+    Do not use a proxy server at all.
+
+`default'
+    Use the value of `url-proxy-services'.
+
+Alist
```

Comment:
Honestly I think a bypass boolean is the way to go right now. The only use case we have for this feature is bypassing proxies, so the ability to specify a completely alternative list feels like feature bloat. So probably something like `ycmd-bypass-url-proxy-services` which defaults to true.